flask_rebar/swagger_generation/authenticator_to_swagger.py

```diff
@@ -0,0 +1,168 @@
+from collections import namedtuple
+
+from flask_rebar.utils.deprecation import deprecated, deprecated_parameters
+from flask_rebar.authenticators import HeaderApiKeyAuthenticator, Authenticator
+from .marshmallow_to_swagger import ConverterRegistry
+from . import swagger_words as sw
+
+
+_Context = namedtuple(
+    "_Context",
+    [
+        # The major version of OpenAPI being converter for
+        "openapi_version"
+    ],
+)
+
+
+class AuthenticatorConverter(object):
+    """
+    Abstract class for objects that convert Authenticator objects to
+    security JSONSchema.
+    """
+
+    AUTHENTICATOR_TYPE = None
+
+    def get_security_schemes(self, obj, context):
+        """
+        Get the security schemes for the provided Authenticator object
+        :param flask.authenticators.Authenticator obj:
+        :param _Context context:
+        :rtype: dict
+        """
+        raise NotImplementedError()
+
+    def get_security_requirements(self, obj, context):
```

Comment:
It took me a while to understand what you meant by requirements. I believe we should use `get_security_scopes` to stay consistent with the openapi documentation. Maybe also reference the part of their doc for that.

Reply:
Hmm, that's not good. 
I was trying to be consistent with the OpenApi documentation names, but the purpose of the methods needs to be as clear as possible.

Relevant parts I looked at:

> Definition/Scheme:
>https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#security-definitions-object
>https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#security-scheme-object
>https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#security-scheme-object

>Requirements:
>https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#security-requirement-object
>https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#security-requirement-object

@Sytten Do you think adding links  to reference those parts of the docs would be enough to remove confusion?

Reply:
Yes, especially since we want users to extend those classes. I was using this doc: https://swagger.io/docs/specification/authentication/#scopes which doesn't mention the word security requirements anywhere :S